src/go/rpk/pkg/cli/container/start.go

```diff
@@ -119,17 +173,43 @@ You can retry profile creation by running:
 	command.Flags().StringVar(&image, "image", common.DefaultImage(), "An arbitrary container image to use")
 	command.Flags().BoolVar(&pull, "pull", false, "Force pull the container image used")
 	command.Flags().BoolVar(&noProfile, "no-profile", false, "If true, rpk will not create an rpk profile after creating a cluster")
+	command.Flags().StringSliceVar(&kPorts, flagKafkaPorts, nil, "Kafka protocol ports to listen on; check help text for more information")
+	command.Flags().StringSliceVar(&aPorts, flagAdminPorts, nil, "Redpanda Admin API ports to listen on; check help text for more information")
+	command.Flags().StringSliceVar(&srPorts, flagSRPorts, nil, "Schema registry ports to listen on; check help text for more information")
+	command.Flags().StringSliceVar(&pPorts, flagProxyPorts, nil, "Pandaproxy ports to listen on; check help text for more information")
+	command.Flags().StringSliceVar(&rPorts, flagRPCPorts, nil, "RPC ports to listen on; check help text for more information")
+	// opt-in for 'any' in all listeners
+	command.Flags().BoolVar(&anyPort, flagAnyPort, false, "Opt in for any (random) ports in all listeners")
+
+	command.MarkFlagsMutuallyExclusive(flagAnyPort, flagKafkaPorts)
+	command.MarkFlagsMutuallyExclusive(flagAnyPort, flagAdminPorts)
+	command.MarkFlagsMutuallyExclusive(flagAnyPort, flagSRPorts)
+	command.MarkFlagsMutuallyExclusive(flagAnyPort, flagProxyPorts)
+	command.MarkFlagsMutuallyExclusive(flagAnyPort, flagRPCPorts)
 
 	return command
 }
 
+func collectFlags(args []string, flag string) []string {
```

Comment:
This is fine (and is just moving code), but I wonder if our cobraext.LongFlagValue func would be better to use

Reply:
I don't think is good for this particular usage:

`collectFlags` here is used to collect the values (`[]string`) of the `--set` flag, which is not part of the FlagSet of the command. cobraext.LongFlagValue collect single-value flags and it collects them from flags that belong to the FlagSet